src/components/course/tests/CourseSkills.test.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { AppContext } from '@edx/frontend-platform/react';
+import { screen, render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import { CourseContextProvider } from '../CourseContextProvider';
+import CourseSkills from '../CourseSkills';
+
+import { SKILLS_BUTTON_TEXT } from '../data/constants';
+
+/* eslint-disable react/prop-types */
+const CourseSkillsWithContext = ({
+  initialAppState = {},
+  initialCourseState = {},
+}) => (
+  <AppContext.Provider value={initialAppState}>
+    <CourseContextProvider initialState={initialCourseState}>
+      <CourseSkills />
+    </CourseContextProvider>
+  </AppContext.Provider>
+);
+/* eslint-enable react/prop-types */
+
+describe('<CourseSkills />', () => {
+  const initialAppState = {
+    enterpriseConfig: {
+      slug: 'test-enterprise-slug',
+    },
+  };
+  const initialCourseState = {
+    course: {
+      skillNames: ['test-skill1', 'test-skill2', 'test-skill3', 'test-skill4'],
+    },
+  };
+
+  test('renders course skills less than limit', () => {
+    render(
+      <CourseSkillsWithContext
+        initialAppState={initialAppState}
+        initialCourseState={initialCourseState}
+      />,
+    );
+    initialCourseState.course.skillNames.forEach((skill) => {
+      expect(screen.queryByText(skill)).toBeVisible();
+    });
+  });
+
+  test('renders course skills greater than limit', () => {
+    const skillGreaterThanLimit = ['test-skill1', 'test-skill2', 'test-skill3', 'test-skill4', 'test-skill5', 'test-skill6'];
+    const courseState = {
+      ...initialCourseState,
+      course: {
+        ...initialCourseState.course,
+        skillNames: skillGreaterThanLimit,
+      },
+    };
+    render(
+      <CourseSkillsWithContext
+        initialAppState={initialAppState}
+        initialCourseState={courseState}
+      />,
+    );
+    initialCourseState.course.skillNames.forEach((skill, index) => {
+      if (index >= 5) {
```

Comment:
SUGGEST: Since this number 5 is basically part of product business logic/spec, I would make this an imported constant from the CourseSkills component since then it's one place to change it and tests are still stable

Reply:
Totally agree, sorry I forgot to change this bit.